recurly/notifications/notifications.go

```diff
@@ -0,0 +1,132 @@
+package notifications
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/blacklightcms/go-recurly/recurly"
+)
+
+// Webhook notification constants.
+const (
+	SuccessfulPayment = "successful_payment_notification"
+	FailedPayment     = "failed_payment_notification"
+	PastDueInvoice    = "past_due_invoice_notification"
+)
+
+type notificationName struct {
+	XMLName xml.Name
+}
+
+type (
+	// SuccessfulPaymentNotification is sent when a payment is successful.
+	SuccessfulPaymentNotification struct {
+		Account       recurly.Account
+		Transaction   recurly.Transaction
+		InvoiceNumber int `xml:"-"`
+	}
+
+	// FailedPaymentNotification is sent when a payment fails.
+	FailedPaymentNotification struct {
+		Account       recurly.Account
+		Transaction   recurly.Transaction
+		InvoiceNumber int `xml:"-"`
+	}
+
+	// PastDueInvoiceNotification is sent when an invoice is past due.
+	PastDueInvoiceNotification struct {
+		Account recurly.Account
+		Invoice recurly.Invoice
+	}
+)
+
+// transactionHolder allows the uuid and invoice number fields to be set.
+// The UUID field is labeled id in notifications and the invoice number
+// is not included on the existing transaction struct.
+type transactionHolder interface {
+	setTransactionFields(id string, in int)
+}
+
+// setTransactionFields sets fields on the transaction struct.
+func (n *SuccessfulPaymentNotification) setTransactionFields(id string, in int) {
+	n.Transaction.UUID = id
+	n.InvoiceNumber = in
```

Comment:
With the latest version on `master`, you can now set `n.Transaction.InvoiceNumber` and remove the `InvoiceNumber` from each of the `Notification` structs.